packages/components/src/Box.js

```diff
@@ -1,40 +1,64 @@
-import styled from '@emotion/styled'
+/** @jsx jsx */
+import { jsx, useTheme } from '@emotion/react'
+import { forwardRef } from 'react'
 import { css, get } from '@theme-ui/css'
-import { createShouldForwardProp } from '@styled-system/should-forward-prop'
 import space from '@styled-system/space'
 import color from '@styled-system/color'
 
 const boxSystemProps = [...space.propNames, ...color.propNames]
-
 /**
  * @internal
  * @type {(prop: string) => boolean}
  */
 export const __isBoxStyledSystemProp = (prop) => boxSystemProps.includes(prop)
 
-const shouldForwardProp = createShouldForwardProp(boxSystemProps)
-
 const sx = (props) => css(props.sx)(props.theme)
 const base = (props) => css(props.__css)(props.theme)
 const variant = ({ theme, variant, __themeKey = 'variants' }) =>
-  css(get(theme, __themeKey + '.' + variant, get(theme, variant)))
+  css(get(theme, __themeKey + '.' + variant, get(theme, variant)))(theme)
 
-export const Box = styled('div', {
-  shouldForwardProp,
-})(
-  {
-    boxSizing: 'border-box',
-    margin: 0,
-    minWidth: 0,
-  },
-  base,
-  variant,
-  space,
-  color,
-  sx,
-  (props) => props.css
-)
+const objToArray = (obj) =>
+  obj ? Object.keys(obj).map((key) => ({ [key]: obj[key] })) : []
 
-Box.displayName = 'Box'
+const mergeProps = (props, initial, ...args) => {
+  return args.reduce(
+    (acc, fn) => [...acc, ...objToArray(fn(props))],
+    objToArray(initial)
+  )
+}
+export const Box = forwardRef(function Box(props, ref) {
+  const theme = useTheme()
+  const {
+    variant: variantProp,
+    __themeKey = 'variants',
+    __css,
+    css: cssProp,
+    sx: sxProp,
+    as: Component = 'div',
+    ...rest
+  } = props
+  const style = mergeProps(
+    { theme, ...props },
+    {
+      boxSizing: 'border-box',
+      margin: 0,
+      minWidth: 0,
+    },
+    base,
+    variant,
+    space,
+    color,
+    sx,
+    (props) => props.css
+  )
+  boxSystemProps.forEach((name) => {
+    delete rest[name]
+  })
+  return <Component ref={ref} css={style} {...rest} />
+})
 
+Box.withComponent = (component) => ({ as, ...props }) => {
```

Comment:
This is actually still breaking.

Previously you could repeat it as much as you want.
```
Box.withComponent('div').withComponent('form')
```

I'm not really opinionated about it, so it's not a suggestion — I just wanted to let you know.

Reply:
yeah, we marked it as deprecated anyway :)

Reply:
Personally I'm fine with this — this was never a documented feature anyway. Did we want to add a note to the deprecation warning?